lib/tasks/fake_data.rake

```diff
@@ -216,11 +216,59 @@ namespace :fake_data do
     end
   end
 
+  desc 'create fake journeys'
+  task create_journeys: :environment do
+    puts 'create_journeys...'
+    Move
+        .left_outer_joins(:journeys).where(journeys: { move_id: nil })
+        .where(status: %w[completed booked requested])
+        .find_each do |move|
+      # Creates some random journeys between A and B
+      current_location = move.from_location
+      number_of_journeys = rand(1..5)
+      journey_counter = 1
+      supplier = move.from_location.suppliers.first || Supplier.all.sample # pick a supplier based on the from location or at random
+      to_location = move.to_location || Location.order('RANDOM()').where(location_type: 'prison').first # pick a random prison if to_location was not specified
+      timestamp = move.date + rand(5..10).hours
+      vehicle = {
+        "id": rand(1..9999).to_s,
+        "registration": "#{('A'..'Z').to_a.sample(3).join}-#{rand(1..100)}",
+      }
+
+      while journey_counter < number_of_journeys
+        intermediate_location = Location.where.not(id: [move.from_location.id, current_location.id, to_location.id]).order('RANDOM()').first
+
+        move.journeys.create!(
+          client_timestamp: timestamp,
+          supplier: supplier,
+          from_location: current_location,
+          to_location: intermediate_location,
+          state: 'completed',
+          billable: [true, true, true, false].sample,
+          vehicle: vehicle,
+        )
+
+        timestamp += rand(30..90).minutes
+        current_location = intermediate_location
+        journey_counter += 1
+      end
+      move.journeys.create!(
+        client_timestamp: timestamp,
+        supplier: supplier,
+        from_location: current_location,
+        to_location: to_location,
+        state: 'completed',
```

Comment:
Just wondering about moves in the current states: `booked, requested` how does a journey be completed if they are still requested, maybe booked is ok? Can we only set those journeys on completed moves instead?

Reply:
Unfortunately, none of the moves created through the front end get beyond the requested stage (as suppliers are not yet fully integrated with us). So this will create simulated journeys for those records which aren't completed but probably should have been.